### PR TITLE
Refactor: Simplify and clean up the code

### PR DESCRIPTION
**Pull request by eumiro**
Several independent refactoring improvements simplifying and cleaning up the code.

### CODE REVIEW

File: src/tox/config/of_type.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 """
 Group together configuration values that belong together (such as base tox configuration, tox environment configs)
 """
+import itertools as it
```
Comment (gaborbernat) [inline src/tox/config/of_type.py:4]: `from itertools import product`

File: tests/session/cmd/test_sequential.py [hunk 2]
```diff
@@ -88,15 +88,15 @@ def test_result_json_sequential(
     packaging_test = get_cmd_exit_run_id(log_report, ".pkg", "test")
     assert packaging_test == [(None, "build_wheel")]
     packaging_installed = log_report["testenvs"][".pkg"].pop("installed_packages")
-    assert {i[: i.find("==")] for i in packaging_installed} == {"pip", "setuptools", "wheel"}
+    assert {i.partition("==")[0] for i in packaging_installed} == {"pip", "setuptools", "wheel"}
```
Comment (gaborbernat) [inline tests/session/cmd/test_sequential.py:91]: In these two cases here I find `find` easier to read and understand than partition 🤔

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/tox/config/of_type.py (hunk 1 above)
Comment (eumiro) [reply src/tox/config/of_type.py:4]: Changed. How about importing from `itertools` in `src/tox/config/loader/ini/factor.py`?

File: tests/session/cmd/test_sequential.py (hunk 2 above)
Comment (eumiro) [reply tests/session/cmd/test_sequential.py:91]: First, I experimented with `assert all(i.startswith(("pip==", "setuptools==", "wheel==")) for i in packaging_installed)` but `pytest` would probably won't be very helpful if the test failed. I'll revert my change.

File: src/tox/config/of_type.py (hunk 1 above)
Comment (gaborbernat) [reply src/tox/config/of_type.py]: What do you mean?

**Review by gaborbernat — APPROVED** [review]
Given all CI will pass.

File: src/tox/config/of_type.py (hunk 1 above)
Comment (eumiro) [reply src/tox/config/of_type.py]: Everywhere else there's `from itertools import …`, only in that `src/tox/config/loader/ini/factor.py` there's the plain `import itertools`. I'm asking whether you'd like to have it repo-wide consistent.

Comment (gaborbernat) [reply src/tox/config/of_type.py]: Oh must be an oversight, definitely let's keep it consistent.👌

Comment (eumiro) [reply src/tox/config/of_type.py]: Okay, will add an extra commit to this PR.

**Pull request merged**